lookout/style/format/quality_report_noisy.py

```diff
@@ -262,59 +266,80 @@ def quality_report_noisy(true_repo: str, noisy_repo: str, bblfsh: str, language:
-    vnodes_y_true = files2vnodes(true_files, feature_extractor, client)
-    mispreds_noise = files2mispreds(noisy_files, feature_extractor, rules, client, log)
-    diff_mispreds = get_diff_mispreds(mispreds_noise, start_changes)
-    changes_count = len(start_changes)
-
-    precisions, recalls = [], []
-    n_rules = len(rules.rules)
-    rules_id = [(i, r.stats.conf) for i, r in enumerate(rules.rules)
-                if r.stats.conf > confidence_threshold and r.stats.support > support_threshold]
-    rules_id = sorted(rules_id, key=lambda k: k[1], reverse=True)
-    for i in range(len(rules.rules)):
-        filtered_mispreds = {k: m for k, m in diff_mispreds.items()
-                             if any(r[0] == m.rule for r in rules_id[:i + 1])}
-        style_fixes = get_style_fixes(filtered_mispreds, vnodes_y_true,
-                                      true_files, noisy_files, feature_extractor)
-        precision, recall, f1_score = compute_metrics(changes_count=changes_count,
-                                                      predictions_count=len(filtered_mispreds),
-                                                      true_positive=len(style_fixes))
-        precisions.append(round(precision, 3))
-        recalls.append(round(recall, 3))
-    print("recall x:", recalls)
-    print("precision y:", precisions)
-
-    # compute some stats and quality metrics for the model's evaluation
-    n_mistakes = len(true_files)
-    prec_max_rec = precisions[-1]
-    max_rec = max(recalls)
-    n_rules_filtered = len(rules_id)
-    # compute the confidence and recall limit for the given precision threshold
-    for i, (prec, rec) in enumerate(zip(precisions, recalls)):
-        if prec < precision_threshold:
-            break
-        confidence_threshold_exp = round(rules.rules[i].stats.conf, 3)
-        rec_threshold_prec = rec
-
-    # compile the precision-recall curve
-    path_to_figure = os.path.join(dir_output, "pr_curve_jquery.png")
-    plot_curve("jquery/jquery", numpy.asarray(recalls), numpy.asarray(precisions),
-               precision_threshold, rec_threshold_prec, confidence_threshold_exp, path_to_figure)
+    repositories = []
+    precisions, recalls = (defaultdict(list) for _ in range(2))
+    n_mistakes, rec_threshold_prec, prec_max_rec, confidence_threshold_exp, max_rec, \
+        n_rules, n_rules_filtered = ({} for _ in range(7))
+    for url in REPOSITORIES.split("\n"):
+        repo = url.split("/")[-1]
+        log.info("Fetching %s", url)
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            git_dir = os.path.join(tmpdirname, repo)
+            git_dir_noisy = os.path.join(tmpdirname, repo + "_noisy")
+            cmd1 = "git clone --single-branch --branch master %s %s" % (url, git_dir)
+            cmd2 = "git clone --single-branch --branch style-noise-1-per-file %s %s" % (
+                    url, git_dir_noisy)
+            process = subprocess.Popen("%s; %s" % (cmd1, cmd2), shell=True)
```

Comment:
Please use subprocess.check_call and also do two commands independently without using a shell